posthog/warehouse/api/data_warehouse.py

```diff
@@ -178,3 +182,147 @@ def recent_activity(self, request: Request, **kwargs) -> Response:
                 "previous": prev_url,
             }
         )
+
+    @action(methods=["GET"], detail=False)
+    def job_stats(self, request: Request, **kwargs) -> Response:
+        """
+        Returns success and failed job statistics for the last 1, 7, or 30 days.
+        Query parameter 'days' can be 1, 7, or 30 (default: 7).
+        """
+
+        # TODO: Will want to cache this data to not spam PG
+        try:
+            days = int(request.GET.get("days", 7))
+            if days not in [1, 7, 30]:
+                return Response(
+                    {"error": "Invalid days parameter. Must be 1, 7, or 30."},
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+        except (ValueError, TypeError):
+            return Response({"error": "Invalid days parameter"}, status=status.HTTP_400_BAD_REQUEST)
+
+        project_tz = ZoneInfo(self.team.timezone) if self.team.timezone else ZoneInfo("UTC")
+        now = datetime.now(project_tz)
+        cutoff_time = now - timedelta(days=days)
+
+        try:
+            external_jobs = ExternalDataJob.objects.filter(team_id=self.team_id, created_at__gte=cutoff_time)
```

Comment:
You'll wanna add `billable=True` here and `finished_at__isnull=False` to make sure you get only finished jobs (or dont use `created_at`, but use `finished_at` instead - this is whats used in billing)